### DISCUSSION TITLE
Will the p99 time-consuming include the time-consuming of failure or timeout?

### DISCUSSION THREAD

**Discussion by hzyforme** [apache/skywalking#11833] [Q&A] [answered]
Assuming that timeout is configured and the number of timeout requests is only 0.1%, will it affect the overall p99 time-consuming? How to calculate the p99 time-consuming for timeout requests and normal requests? Should p99 be calculated separately for timed-out requests and normal requests, or should p99 be calculated for all requests together?

**Comment by wu-sheng** [comment, marked as answer]
I am not following about `should`.  If you are asking, by default is SkyWalking considering a timeout as a part of the latency statistics? Of course. I would follow up with a question for you, `why not`?

If you want to exclude that, sure, it is possible, you could use OAL(if the data is from tracing or mesh), to filter the error status request out, and then do `percentile`.
